### PR TITLE
Check the scalar value of an ECPrivateKey

### PR DESCRIPTION
**Pull request by JinhangZhang**
OpenJCEPlus/OpenJCEPlusFIPS providers accepts an invalid EC private key at the moment. Need to add a check to ensure the EC private key is valid during the initialization procedure in key agreement and ecdsa signature.

cherry-pick from https://github.com/IBM/OpenJCEPlus/pull/190

### CODE REVIEW

File: src/main/java/com/ibm/crypto/plus/provider/ECUtils.java [hunk 1]
```diff
@@ -254,4 +257,44 @@ static String bytesToHex(byte[] bytes) {
         return new String(hexChars);
     }
 
+     /**
```
Comment (jasonkatonica) [inline src/main/java/com/ibm/crypto/plus/provider/ECUtils.java:260]: Indentation here is slightly different then the Java 17 PR. Please make this match.

File: src/main/java/com/ibm/crypto/plus/provider/ECUtils.java [hunk 2]
```diff
@@ -254,4 +257,44 @@ static String bytesToHex(byte[] bytes) {
...
+                        sVal.compareTo(order) >= 0) {
+                    throw new InvalidKeyException("The private key must be " +
+                            "within the range [1, n - 1]");
+                }
+            }
+        }
+
+        return prv;
+    }
+
```
Comment (jasonkatonica) [inline src/main/java/com/ibm/crypto/plus/provider/ECUtils.java:299]: An additional blank line is here please make this match.

**Review by jasonkatonica — CHANGES_REQUESTED** [review]

**Review by johnpeck-us-ibm — APPROVED** [review]

File: src/main/java/com/ibm/crypto/plus/provider/ECUtils.java (hunk 1 above)
Comment (JinhangZhang) [reply src/main/java/com/ibm/crypto/plus/provider/ECUtils.java:260]: updated

File: src/main/java/com/ibm/crypto/plus/provider/ECUtils.java (hunk 2 above)
Comment (JinhangZhang) [reply src/main/java/com/ibm/crypto/plus/provider/ECUtils.java:299]: updated

**Review by KostasTsiounis — APPROVED** [review]
LGTM

**Comment by jasonkatonica** [comment]
x86_64_linux, ppc64le_linux, s390x_linux, x86_64_windows

**Review by jasonkatonica — APPROVED** [review]
LGTM

**Pull request merged**